### ISSUE TITLE
Testcafe breaks XHR when using createObjectURL

### ISSUE THREAD

**Issue by mixtur** [DevExpress/testcafe#6276]
### What is your Test Scenario?
Test this page
```html
<script>
    const data = new Uint8Array([1, 2, 3, 4]);
    const url = URL.createObjectURL(new Blob(data));
    const xhr = new XMLHttpRequest();
    xhr.open('GET', url);
    xhr.onload = () => {
        console.log(123);
    };
    xhr.send();
</script>
```
 
using this code

```js
fixture('testest')
    .page('./index.html');//when using server it fails all the same, so it is ok

test('Test 0', async t => {
});
```

### What is the Current behavior?
This error
```
hammerhead.js:14 Uncaught TypeError: Cannot read property 'withCredentials' of undefined
    at XMLHttpRequest.send (hammerhead.js:14)
    at index.html:10
send @ hammerhead.js:14
(anonymous) @ index.html:10
``` 
### What is the Expected behavior?
no error and `123` logged in console

### Steps to Reproduce:
Run the test against the page.
 
### Your Environment details:
 
* testcafe version:                   1.14.2
* node.js version:                    14.17.0
* command-line arguments:  `testcafe chrome testest.js --debug-on-fail`
* browser name and version: Chrome 91.0.4472.77 / Linux 0.0
* platform and version:          Ubuntu 20.04.2 LTS

**Comment by mixtur** [comment]
Also everything was ok in `testcafe@1.10.1` so the bug must have been introduced somewhere between 1.10.1 and  1.14.2

**Comment by Dmitry-Ostashev** [comment]
Thank you for your report. I reproduced the issue. We'll research it and update this thread once we have any news.

**Comment by msherman** [comment]
Are there any updates to this issue?

**Comment by mixtur** [comment]
It's been half of a year now. Just saying.

**Comment by Aleksey28** [comment]
Hi @mixtur,

We fixed the issue in version 1.18.5. I checked this with the latest Testcafe versions (1.18.5, 1.18.6, and 1.19.0) and it works fine. So, please update to one of these versions to resolve the issue.

**Comment by mixtur** [comment]
Thank you :)
